src/pykeen/evaluation/evaluator.py

```diff
@@ -186,9 +186,10 @@ def evaluate(
 
         if batch_size is None and self.automatic_memory_optimization:
             # Using automatic memory optimization on CPU may result in undocumented crashes due to OS' OOM killer.
-            if model.device.type == "cpu":
+            if model.device.type == "cpu" or model.device.type == "mps":
```

Comment:
Would this also work?
```suggestion
            if model.device.type != "cuda":
```

Reply:
Generally yes, trying out moving tensors to mps returns respectable results:
```bash
>>> a = torch.rand(5,5,device="mps")
>>> a.device.type
'mps'
```

Reply:
Can we update this logger.info to better explain why this only works on cuda GPUs?

Reply:
FYI this line of code has been upstreamed into max's AMO library

Reply:
(the warning part of it 😄 we still need a distinction here to avoid OOM killer action for non-cuda parts)